plugins/auth-backend-module-microsoft-provider/src/authenticator.ts

```diff
@@ -22,6 +22,8 @@ import {
   PassportProfile,
 } from '@backstage/plugin-auth-node';
 
+let domainHint: string | undefined = undefined;
```

Comment:
No globals please, basically ever :) Let's find some alternative.

One way:

```ts
const helper = PassportOAuthAuthenticatorHelper.from(...);
helper.domainHint = domainHint;
```

and then you can access `(this as { domainHint?: string }).domainHint` or so in the start method.

Or:

When calling `done` below, maybe you have access to the options in `params`? Then that's an opportunity to inject this extra data.

Reply:
Taking a second look, the second method to `start` is a generic `ctx` parameter, rather than specifically a passport authenticator helper object.  That makes it nice and easy to pass a wrapper object containing the hint and the helper.

Much nicer!